common/src/main/scala/net/psforever/packet/game/objectcreate/CharacterData.scala

```diff
@@ -164,7 +159,23 @@ object CharacterAppearanceData extends Marshallable[CharacterAppearanceData] {
       ("unk8" | uint4L) ::
       ignore(6) ::
       ("ribbons" | RibbonBars.codec)
-    ).as[CharacterAppearanceData]
+    ).exmap[CharacterAppearanceData] (
+    {
+      case a :: _ :: b :: _ :: c :: d :: e :: _ :: f :: g :: _ :: h :: i :: j :: k :: l :: _ :: m :: n :: o :: _ :: p :: _ :: q :: _ :: r :: s :: t :: _ :: u :: HNil =>
+        Attempt.successful(
+          CharacterAppearanceData(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u)
+        )
+    },
+    {
+      case CharacterAppearanceData(_, _, PlanetSideEmpire.NEUTRAL, _, _, name, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _) =>
+        Attempt.failure(Err(s"character $name's faction can not declare as neutral"))
+
+      case CharacterAppearanceData(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u) =>
+        Attempt.successful(
+          a :: () :: b :: () :: c :: d :: e :: () :: f :: g :: () :: h :: i :: j :: k :: l :: () :: m :: n :: o :: () :: p :: () :: q :: () :: r :: s :: t :: () :: u :: HNil
```

Comment:
Yep, hell.  And that's still ignoring about half of that part of the packet.

The good news is that we're going to have to split this `case class` the better we understand the individual fields anyway and make other `case class`es to compartmentalize fields.  They will be smaller once we have better, more logical associations with the packet data.